### PR TITLE
Add genotyping and filtering option to single-sample workflow

### PR DESCRIPTION
**Pull request by ldgauthier**
I have a version without imports that works.  This version won't validate with womtool, probably because of the imports:
`ERROR: Cannot find reference to 'CheckContamination' for member access 'CheckContamination.contamination' (line 190, col 47):`

The accuracy for the synthetic diploid sample (https://doi.org/10.1101/223297) was compared with a ~800 sample callset from production the sensitivity is very close for SNPs (slightly lower), better sensitivity for indels, and more FPs for SNPs and indels.  

97.3% sensitivity for SNPs
99.4% precision for SNPs
65.9% sensitivity for indels*
98.7% precision for indels -- 3.8FP/Mb

*This dataset excludes 1bp indels, which are the most common and also the easiest.  It also includes some very large events from PacBio not possible to call with Illumina.

SNP sensitivity is on par with that reported in the SynDip paper. 
Indel sensitivity is lower.

### CODE REVIEW

**Comment by ldgauthier** [comment]
Latest update is working with and without genotyping and filtering.  I effectively overrode some of the existing tasks, which is maybe not good WDL style, but I needed new arguments.

For the record, the directory structure of the imports is a big pain for users (like me) running Cromwell in server mode: https://gatkforums.broadinstitute.org/gatk/discussion/comment/46211
It would be great if all the imports were inside some parent folder that could be easily zipped to submit the subworkflows.

**Comment by bshifaw** [comment]
The imports structure shouldn't be a problem to change for this repo since there's only one main workflow calling the tasks. But this will probably continue in dsde pipelines where there maybe different workflows calling the same tasks. 
So I'll have all imports under a directory called subworkflows and leave the main workflows and json as is. 
|_README.md	
|_germline_single_sample_workflow.hg38.inputs.json	
|_germline_single_sample_workflow.wdl	
|_subworkflows
|.....|_split_large_readgroup.wdl	
|.....|_unmapped_bam_to_aligned_bam.wdl
|.....|_alignment.wdl	
|.....|_bam_processing.wdl	
|.....|_germline_variant_discovery.wdl	
|.....|_qc.wdl	
|.....|_utilities.wdl

File: germline_single_sample_workflow.wdl [hunk 1]
```diff
@@ -297,3 +321,125 @@ workflow germline_single_sample_workflow {
   }
 }
 
+# Call variants on a single sample with HaplotypeCaller to produce a GVCF
+task HaplotypeCaller {
```
Comment (bshifaw) [inline germline_single_sample_workflow.wdl:325]: So the changes to Haplotypecaller and FilterVCF would go in germline_variant_discovery.wdl

File: germline_single_sample_workflow.wdl [hunk 2]
```diff
@@ -297,3 +321,125 @@ workflow germline_single_sample_workflow {
...
+  runtime {
+    preemptible: preemptible_tries
+    memory: "3500 MB"
+    disks: "local-disk " + sub(disk_size, "\\..*", "") + " HDD"
+    docker: "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.3-1513176735"
+  }
+}
+
+# Validate a (G)VCF with -gvcf specific validation if necessary
+task ValidateGVCF {
```
Comment (bshifaw) [inline germline_single_sample_workflow.wdl:413]: Changes to be added to qc.wdl

**Review by bshifaw — COMMENTED** [review]
-I was speaking with Jose and i believe the aim is to have the main workflow to be task less. For example the haplotypcaller task would be added to germline_variant_discovery.wdl.
-I'm wondering if these changes should be here or in dsde-pipelines then waterfalled to gatk-workflows and FC. This would keep the wdls consistent instead of having different versions.

**Pull request closed without merging**